dataland-frontend/src/utils/Email.ts

```diff
@@ -0,0 +1,17 @@
+import { type Card } from "@/types/ContentTypes";
+
+/**
+ * Opens the email client with an email generated from the information provided by a content card
+ * @param card the card providing the content
+ */
+export function openEmailClient(card?: Card): void {
+  if (card) {
+    const email = card?.icon ?? "";
+    const subject = card?.title ?? "";
+    const body = card?.text ?? "";
+
+    if (email && subject && body) {
+      window.location.href = `mailto:${email}?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;
```

Comment:
I see potential problems with this method if it accidentally does not receive a card with all three required attributes (email, subject, body). Then it will just do nothing.

My suggestion is to pass "email, subject, body" as three params to this method, and make all of them non-nullable.
That would force the developers to assure that the method receives actual values for all three parameters when calling this method.

I have a feeling that this might avoid email buttons to break in the future.

Reply:
Ok if it will be replaced, it is ok for now. Thx 👍